models/dataset.py

```diff
@@ -53,6 +53,37 @@ def end_date(self) -> datetime:
 
         return time_values.max()
 
+    @cached_property
+    def date_cutoff(self) -> datetime:
+        # The cut-off date for dataset train/test split
+        df = self.df
+        dates_unique = df[self.time_column].unique()
+        date_cutoff = sorted(dates_unique)[
+            round(len(dates_unique) * self.train_percentage)
+        ]
+
+        return date_cutoff
+
+    @property
+    def df_train(self) -> pd.DataFrame:
+        df = self.df
+        df[self.time_column] = df[self.time_column].astype("datetime64[ns]")
+
+        # Split dataset based on date cut-off
+        df_train = df[df[self.time_column] < self.date_cutoff]
+
+        return df_train
+
+    @property
+    def df_test(self) -> pd.DataFrame:
+        df = self.df
+        df[self.time_column] = df[self.time_column].astype("datetime64[ns]")
```

Comment:
Move to `df` property.

Reply:
I removed this line from df_train and df_test as df property has the following line which I think does the same thing :-?
`df[self.time_column] = pd.to_datetime(df[self.time_column], utc=True)`